lib/assessment.js

```diff
@@ -8,25 +8,80 @@
 // Question 1
 // Write a callback for reduce that selects
 // the minimum value in the array
-var min = function min() {
 
+var min = function min(previousValue, currentValue, index, array) {
+  return previousValue < currentValue ? previousValue : currentValue;
 };
 
+console.log(reduce(numbers, min));
+console.log(numbers.reduce(function (previousValue, currentValue) {
+  return previousValue < currentValue ? previousValue : currentValue;
+}));
+
+
+};
+
+// I used Antony's Max Example of code and switched the operators.
+// I do not think I could logic out the whole thing as concisely as this.
+// I cannot write this code from scratch myself but I can read this and edit it
+
+
+
 // Question 2
 // Write a callback for reduce that produces
 // the sum of the values in the array
+
 var sum = function sum() {
 
+var sum = function sum(previousValue, currentValue, index, array) {
+  return previousValue + currentValue;
+};
+
+var numbers = [13, 26, 32, 5];
+
+console.log(reduce(numbers, sum));
+console.log(numbers.reduce(function (previousValue, currentValue) {
+  return previousValue + currentValue;
+}));
+
+
 };
 
+// Same case for solving here as above-
+// I heavily relied on Antony's code and can readi it pretty well
+// I Feel rushed so maybe I am not catching all of the edits but
+// I am concerned that I cannot construct this code myself,
+//I hoep i chose the correct code to model from, even.
+// I do understand how javascript is fucntioning though, most of the time.
+
+
+
 // Question 3
 // Write a function that returns true
 // if invoking the callback for any element
 // in the array returns a truthy value
-var any = function any(array, callback, thisArg) {
 
+  var any = function any(array, callback, thisArg) {
+  // callback(currentValue, index, array)
+
+  //check for true first
+  var result = true;
+  array.forEach(function(currentValue, index, array) {
+    // if callback ever returns false, result will be set
+    // to false amd the callback will not be called in any
+    // subsequent iterations.
+    result = result && callback(currentValue, index, array);
+  });
+
+  return result;
 };
```

Comment:
This is the body of Antony's `every` function, which is a good starting point. `any` differs from `every` in that we want one `true` to win out over an indeterminate number of `false`s, whereas `every` wants one `false` to win out over an indeterminate number of `true`s. This means using a different starting value for our boolean accumulator variable and a different logical operation.

Why don't we work through this, and more general problem solving approaches, after 4:30 today.